staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -243,142 +297,473 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 				if err != nil {
-					t.Fatalf("%s: Error creating request: %s", k, err)
+					t.Errorf("failed to create sock5 server: %v", err)
 				}
+				return ts
+			}
 
-				spdyTransport := NewRoundTripper(testCase.clientTLS, redirect, redirect)
+			testCases := map[string]struct {
+				clientTLS              *tls.Config
+				proxyAuth              *url.Userinfo
+				serverConnectionHeader string
+				serverFunc             serverFunc
+				serverStatusCode       int
+				serverUpgradeHeader    string
+				shouldError            bool
+			}{
+				"proxied tcp->tcp via SOCKS5": {
+					serverFunc:             httptest.NewServer,
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					serverUpgradeHeader:    "SPDY/3.1",
+					shouldError:            false,
+				},
+				"proxied tcp->tcp via SOCKS5 with invalid auth": {
+					serverFunc:             httptest.NewServer,
+					proxyAuth:              url.UserPassword("invalid", "auth"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					serverUpgradeHeader:    "SPDY/3.1",
+					shouldError:            true,
+				},
+				"proxied tcp->tcp via SOCKS5 with auth": {
+					serverFunc:             httptest.NewServer,
+					proxyAuth:              url.UserPassword("proxyuser", "proxypasswd"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					serverUpgradeHeader:    "SPDY/3.1",
+					shouldError:            false,
+				},
+				"https (invalid hostname + InsecureSkipVerify)": {
+					serverFunc:             httpsServerInvalidHostname(t),
+					clientTLS:              &tls.Config{InsecureSkipVerify: true},
+					serverConnectionHeader: "Upgrade",
+					serverUpgradeHeader:    "SPDY/3.1",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+				"https (invalid hostname + hostname verification)": {
+					serverFunc:             httpsServerInvalidHostname(t),
+					clientTLS:              &tls.Config{InsecureSkipVerify: false},
+					serverConnectionHeader: "Upgrade",
+					serverUpgradeHeader:    "SPDY/3.1",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            true,
+				},
+				"https (valid hostname + RootCAs)": {
+					serverFunc:             httpsServerValidHostname(t),
+					clientTLS:              &tls.Config{RootCAs: localhostPool},
+					serverConnectionHeader: "Upgrade",
+					serverUpgradeHeader:    "SPDY/3.1",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+				"proxied http->http": {
+					serverFunc:             httptest.NewServer,
+					serverConnectionHeader: "Upgrade",
+					serverUpgradeHeader:    "SPDY/3.1",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+				"proxied https (invalid hostname + InsecureSkipVerify) -> http": {
```

Comment:
because the socks proxy doesn't have a https variant, we don't need the `"proxied https ... -> ...` tests... there's no difference in the proxy we need to exercise

keep the "with auth" testcases, but rename the tests to clarify they are not testing https proxy attributes:

```
proxied with auth -> http
proxied with auth -> https (invalid hostname)
proxied with auth -> https (valid hostname + RootCAs)
```

Reply:
Makes sense, I've removed the https tests and rename all the others, it should be more meaningful like this!